### ISSUE TITLE
Add an Offline Mode

### ISSUE THREAD

**Issue by amdelamar** [desktop/desktop#3854]
### Description

It would be nice to not receive "Failed to fetch" errors when working offline. (Without active internet connection.

![error-failed-to-fetch](https://user-images.githubusercontent.com/18532924/35344258-b04c5852-00fa-11e8-94ac-e6600f7cad71.PNG)

### Version

**GitHub Desktop version:** 1.0.12

**OS version:** Windows 10 - But also not specific to this

### Steps to Reproduce

1. Disconnect WiFi / LAN or enter Airplane Mode
1. View any repository that's fetching origin from GitHub
1. Get an error on opening repo. And every few minutes thereafter. Error: "Failed to fetch"

**Expected behavior:** Gracefully fail to fetch. Please offer a "Go Offline" button in the popup. Offer a "Offline Mode" option in the options menu. And silence repeating errors during fetch if current state is set to "Offline", or don't fetch at all if in offline mode.

**Actual behavior:** Error comes back every few minutes. **Note**: this ends up being very annoying and I close GitHub Desktop and use the command line git whenever I have to work offline.

**Reproduces how often:** 100%

**Comment by iAmWillShepherd** [comment]
Thanks for taking the time to open an issue! 🙇

**Comment by j-f1** [comment]
Maybe we should put an error message near the fetch button instead of opening a dialog.

**Comment by amdelamar** [comment]
I think the user should have more choices.

![go-offline-button](https://user-images.githubusercontent.com/18532924/35345454-ada4dbe4-00fd-11e8-9b91-85dec33277e2.PNG)

They could choose *not* to go offline and receive the same dialog again later, if they wish. But once "offline mode" is enabled, They shouldn't be able to fetch, auto-fetch, or anything. And would expect the "Fetch" button would change to "Go Online and Fetch origin". 

Basically, two modes of operation are needed here. So this is a big change.

**Comment by shiftkey** [comment]
@amdelamar Thanks for the suggestion!

This idea is interesting for the future, but this is beyond the scope of our [current roadmap](https://github.com/desktop/desktop/blob/master/docs/process/roadmap.md).

I added the `future-proposal` label to this issue and closed it so that we won't get overwhelmed with our backlog! We will revisit this in our next planning session.
